### PR TITLE
PS-5651 SpyAvailabilityAbstractTableMap leaks to all entities

### PR DESCRIPTION
**Pull request by m7moud**
- Developer(s): @m7moud

- Ticket: https://spryker.atlassian.net/browse/PS-5651

- Academy PR: -

- Release Group: https://release.spryker.com/release-groups/view/881

#### Please confirm

- [X] No new OS components - or they have been approved by legal department

#### Documentation

- [X] Functional documentation provided or in progress?
- [X] Integration guide for projects provided or not needed?
- [X] Migration guides for all contained majors provided or not needed?

#### Release Table

   Module                | Release Type         | Constraint Updates         |
   :--------------------- | :------------------------ | :--------------------- |
   EventBehavior         | patch                 |                       |

#### Release Notes

Replaced leaking dependency class with same table for same module.

-----------------------------------------

##### Change log

Bugfixes

- Replaced leaking `SpyAvailabilityAbstractTableMap` class with same table for same module.
- Renamed post save and delete methods.

### CODE REVIEW

**Comment by dereuromark** [comment]
requires a release template.

File: src/Spryker/Zed/EventBehavior/Persistence/Propel/Behavior/EventBehavior.php [hunk 1]
```diff
@@ -36,7 +36,7 @@ public function preSave()
     public function postSave()
     {
         return "
-\$this->addSaveEventToMemory();
+\$this->logSavedEventBehaviorEntityChange();
```
Comment (ehsanmx) [inline src/Spryker/Zed/EventBehavior/Persistence/Propel/Behavior/EventBehavior.php]: it's not logging, it's an event entity which will be used for triggering, maybe this one `saveEventBehaviorEntityChange`

Comment (m7moud) [reply src/Spryker/Zed/EventBehavior/Persistence/Propel/Behavior/EventBehavior.php]: There's already `saveEventBehaviorEntityChange` method, but I can change it's name, but what about the post delete method?

Maybe  `saveAddedEventBehaviorEntityChange` and `saveDeleteedEventBehaviorEntityChange`?

File: src/Spryker/Zed/EventBehavior/Persistence/Propel/Behavior/EventBehavior.php [hunk 2]
```diff
@@ -234,6 +244,24 @@ protected function prepareSaveEventName()
         ";
     }
 
+    /**
+     * @return string
+     */
+    protected function addPrepareDeleteEventMethod()
+    {
+        $deleteEvent = 'Entity.' . $this->getTable()->getName() . '.delete';
```
Comment (ehsanmx) [inline src/Spryker/Zed/EventBehavior/Persistence/Propel/Behavior/EventBehavior.php]: can you please replace all string concatenating with sprintf()

File: src/Spryker/Zed/EventBehavior/Persistence/Propel/Behavior/EventBehavior.php [hunk 3]
```diff
@@ -507,7 +508,7 @@ public function addGetPhpType()
  */
 protected function getPhpType(\$xmlValue, \$column)
 {
-    \$columnType = SpyAvailabilityAbstractTableMap::getTableMap()->getColumn(\$column)->getType();
+    \$columnType = $tableMapPhpName::getTableMap()->getColumn(\$column)->getType();
```
Comment (matweew) [inline src/Spryker/Zed/EventBehavior/Persistence/Propel/Behavior/EventBehavior.php]: nice catch

**Pull request merged**